src/app/sign-in/components/form-sign-in/form-sign-in.component.ts

```diff
@@ -92,26 +91,29 @@ export class FormSignInComponent implements OnInit, AfterViewInit {
       Validators.required,
       UsernameValidator.orcidOrEmail,
     ])
-    this.passwordFormControl = new FormControl('', [])
 
     this.authorizationForm = new FormGroup({
-      username: this.usernameFormControl,
-      password: this.passwordFormControl,
+      username: new FormControl(),
+      password: new FormControl(),
       recoveryCode: new FormControl(),
       verificationCode: new FormControl(),
     })
+
+    if (this.email) {
+      this.authorizationForm.addControl('username', this.usernameFormControl)
+      this.authorizationForm.get('username').updateValueAndValidity()
+    }
   }
 
   ngAfterViewInit(): void {
     this.firstInput.nativeElement.focus()
   }
 
   onSubmit() {
-    this.signInLocal.data = this.authorizationForm.getRawValue()
-
-    this.authorizationForm.markAllAsTouched()
+    this.addUsernameValidation()
```

Comment:
** observation no changes require ** 

Addin the validation after submit is a intersting approach, I think I never see before. 
I believe it should work just fine.

an alternative to this is to write our own ErroStateMatcher as we have on the register. https://github.com/ORCID/orcid-angular/blob/master/src/app/register/ErrorStateMatcherForFormLevelErrors.ts

If we ever have any issue with this approach we can do try that.

Reply:
Ohh I see. Ok ok